### PR TITLE
Only validate options that we have types for

### PR DESCRIPTION
**Pull request by Salakar**
Added check to see if key exists in types, otherwise ignore it. No point checking for all types if we only care about a limited set of options.

### CODE REVIEW

**Comment by BridgeAR** [comment]
I would change this to only check own properties. Otherwise I think it's good to check for added attributes. What is your reason for wanting to remove that check?

**Comment by Salakar** [comment]
See commit message for one reason, allows directly passing in class for example that directly implements the required return reply funcs, reduces steps to get the result in this case.

Just one of the steps I took for performance on a client i'm building for educational purposes:
image

File: lib/parser.js [hunk 1]
```diff
@@ -248,7 +248,7 @@ function JavascriptRedisParser (options) {
     throw new TypeError('Please provide all return functions while initiating the parser')
   }
   for (var key in options) {
-    if (typeof options[key] !== optionTypes[key]) {
+    if (optionTypes[key] && typeof options[key] !== optionTypes[key]) {
       throw new TypeError('The options argument contains the property "' + key + '" that is either unkown or of a wrong type')
```
Comment (BridgeAR) [inline lib/parser.js]: Please change the error description accordingly

**Comment by BridgeAR** [comment]
LGTM besides a nit and the test has to be fixed

**Pull request closed without merging**